root/static/js/unsavedChanges.js

```diff
@@ -0,0 +1,21 @@
+const forms = document.getElementsByClassName('confirmIfUnsavedChanges');
+let hasUnsavedChanges = false;
+
+for(const form of forms) {
+    const inputs = Array.prototype.slice.apply(form.getElementsByTagName('input'));
+    const textareas = Array.prototype.slice.apply(form.getElementsByTagName('textarea'));
+    const selects = Array.prototype.slice.apply(form.getElementsByTagName('select'));
+    const elements = inputs.concat(textareas).concat(selects);
+    console.log(elements)
+    for(const elem of elements) {
+        elem.addEventListener('input', () => {if(!hasUnsavedChanges) hasUnsavedChanges = true});
+    }
+    form.addEventListener('submit', () => {if(hasUnsavedChanges) hasUnsavedChanges = false});
+}
+
+window.addEventListener('beforeunload', (e) => {
+    if (hasUnsavedChanges) {
+        (e || window.event).returnValue = '';
```

Comment:
If you have a relevant web link from your research, please included it as a comment. I remember the pain of research and even with the correct answer couldn’t find the spec right now.